bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/api/WebexTeamsApi.java

```diff
@@ -0,0 +1,274 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.webexteams.internal.api;
+
+import static org.openhab.binding.webexteams.internal.WebexTeamsBindingConstants.*;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.util.InputStreamResponseListener;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.webexteams.internal.WebexTeamsConfiguration;
+import org.openhab.binding.webexteams.internal.WebexTeamsHandler;
+import org.openhab.core.config.core.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * WebexTeamsApi implements API integration with Webex Teams.
+ * 
+ * Not using webex-java-sdk since it's not in a public maven repo, and it doesn't easily
+ * support caching refresh tokens between openhab restarts, etc..
+ * 
+ * @author Tom Deckers - Initial contribution
+ * 
+ */
+@NonNullByDefault
+public class WebexTeamsApi {
+
+    private final Logger logger = LoggerFactory.getLogger(WebexTeamsApi.class);
+
+    private final WebexTeamsHandler handler;
+    private final HttpClient httpClient;
+
+    // OAuth members
+    private @Nullable String authToken;
+    private @Nullable String clientId;
+    private @Nullable String clientSecret;
+    private @Nullable String authCode;
+    private @Nullable String redirectUrl;
+    private @Nullable String refreshToken;
+
+    /**
+     * Constructor.
+     * 
+     * @param WebexTeamsHandler hander interact with handler context.
+     */
+    public WebexTeamsApi(WebexTeamsHandler handler, HttpClient httpClient) {
+        this.handler = handler;
+        this.httpClient = httpClient;
+
+        WebexTeamsConfiguration config = handler.getConfigAs(WebexTeamsConfiguration.class);
+        this.authToken = config.token;
+        this.clientId = config.clientId;
+        this.clientSecret = config.clientSecret;
+        this.authCode = config.authCode;
+        this.redirectUrl = config.redirectUrl;
+        this.refreshToken = config.refreshToken;
+
+        try {
+
+            httpClient.start();
```

Comment:
Since your using `getCommonHttpClient` you should call start or stop on the httclient (see javadoc of `getCommonHttpClient`)

Reply:
You mean rather "you should not call... "

Reply:
Done. Removed calls.